File: backend/app/routes/items.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List
from .. import schemas
from ..database import get_db
from ..crud import item as crud_item

router = APIRouter()

@router.get("/", response_model=List[schemas.Item])
def read_items(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    items = crud_item.get_items(db, skip=skip, limit=limit)
    return items

@router.get("/{item_id}", response_model=schemas.Item)
def read_item(item_id: int, db: Session = Depends(get_db)):
    db_item = crud_item.get_item(db, item_id=item_id)
    if db_item is None:
        raise HTTPException(status_code=404, detail="Item not found")
    return db_item

@router.post("/", response_model=schemas.Item, status_code=201)
def create_item(item: schemas.ItemCreate, db: Session = Depends(get_db)):
    return crud_item.create_item(db=db, item=item)

@router.put("/{item_id}", response_model=schemas.Item)
def update_item(item_id: int, item: schemas.ItemUpdate, db: Session = Depends(get_db)):
    db_item = crud_item.update_item(db, item_id=item_id, item=item)
    if db_item is None:
        raise HTTPException(status_code=404, detail="Item not found")
    return db_item

@router.delete("/{item_id}", status_code=204)
def delete_item(item_id: int, db: Session = Depends(get_db)):
    success = crud_item.delete_item(db, item_id=item_id)
    if not success:
        raise HTTPException(status_code=404, detail="Item not found")

File: frontend/src/api/items.ts
import axios from 'axios'
import type { Item, ItemCreate, ItemUpdate } from '../types/item'

const API_URL = 'http://localhost:8000/api/items'

export default {
  async getItems(): Promise<Item[]> {
    const response = await axios.get<Item[]>(API_URL)
    return response.data
  },

  async getItem(id: number): Promise<Item> {
    const response = await axios.get<Item>(`${API_URL}/${id}`)
    return response.data
  },

  async createItem(item: ItemCreate): Promise<Item> {
    const response = await axios.post<Item>(API_URL, item)
    return response.data
  },

  async updateItem(id: number, item: ItemUpdate): Promise<Item> {
    const response = await axios.put<Item>(`${API_URL}/${id}`, item)
    return response.data
  },

  async deleteItem(id: number): Promise<void> {
    await axios.delete(`${API_URL}/${id}`)
  }
}
